tools/python/util/qdq_helpers/qdq_model_utils.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import onnx
+from ..onnx_model_utils import get_producer_consumer_maps, iterate_graph_per_graph_func
+
+
+def _duplicate_dq_nodes_with_multiple_consumers(graph: onnx.GraphProto, **kwargs):
+    updated_graphs = kwargs['updated_graphs']
+    node_to_consumers = kwargs['node_to_consumers']
+    validate_updates = kwargs['validate_updates']
+
+    nodes_to_update = []
+    for node in filter(lambda node: node.op_type == 'DequantizeLinear', graph.node):
+        # node providing graph output won't have consumer nodes
+        consumers = node_to_consumers[node] if node in node_to_consumers else []
+        if len(consumers) > 1:
+            if not all(consumer in graph.node for consumer in consumers):
+                # TODO: If this does ever occur, as long as it's only consumed in one subgraph we could leave that
+                # value as is (no need to handle recursing into the subgraph) and update the consumers in this
+                # graph only
+                raise IndexError("DequantizeLinear node output is consumed by a subgraph. "
+                                 "This is not currently supported.")
+
+            nodes_to_update.append(node)
+
+    if validate_updates:
+        if nodes_to_update:
+            # internal error. we somehow missed an update in the first pass when validate_upates was false
+            raise ValueError('Graph still has DequantizeLinear nodes with multiple consumers.')
+
+        return
+
+    if nodes_to_update:
+        dup_idx = 0
+        new_graph = onnx.GraphProto()
+        graph_outputs = set([output.name for output in graph.output])
```

Comment:
Do we really need a new graph, or can we just create a list of Nodes and run old_graph.node.extend(new_nodes)?

Reply:
Yes. It has to be topologically sorted according to the ONNX spec so we can't just extend.